### ISSUE TITLE
Bogus RCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE warning reported [TEST CASE]

### ISSUE THREAD

**Issue by archiecobbs** [spotbugs/spotbugs#1899]
SpotBugs is generating a bogus `RCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE` warning in the example below.

`pom.xml`:

```xml
<?xml version="1.0" encoding="UTF-8"?>

<project
  xmlns="http://maven.apache.org/POM/4.0.0"
  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
    <modelVersion>4.0.0</modelVersion>
    <groupId>com.example</groupId>
    <artifactId>bug</artifactId>
    <name>Bug Example</name>
    <version>1.0</version>
    <packaging>jar</packaging>
    <properties>

        <!-- Basic stuff -->
        <compiler.source>1.8</compiler.source>
        <compiler.target>1.8</compiler.target>
        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>

        <!-- Dependency versions -->
        <slf4j.version>1.7.30</slf4j.version>
        <maven.compiler.plugin.version>3.7.0</maven.compiler.plugin.version>
        <spotbugs.maven.plugin.version>4.5.0.0</spotbugs.maven.plugin.version>
        <spring.version>5.2.5.RELEASE</spring.version>

    </properties>

    <dependencies>

        <!-- Spring -->
        <dependency>
            <groupId>org.springframework</groupId>
            <artifactId>spring-web</artifactId>
            <version>${spring.version}</version>
        </dependency>

        <!-- Logging -->
        <dependency>
            <groupId>org.slf4j</groupId>
            <artifactId>slf4j-api</artifactId>
            <version>${slf4j.version}</version>
        </dependency>
    </dependencies>

    <build>
        <plugins>

            <!-- Compile -->
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-compiler-plugin</artifactId>
                <version>${maven.compiler.plugin.version}</version>
                <configuration>
                    <source>${compiler.source}</source>
                    <target>${compiler.target}</target>
                    <showWarnings>true</showWarnings>
                    <compilerArgs>
                        <arg>-Xlint:all,-processing</arg>
                        <arg>-Werror</arg>
                    </compilerArgs>
                </configuration>
            </plugin>

            <!-- SpotBugs -->
            <plugin>
                <groupId>com.github.spotbugs</groupId>
                <artifactId>spotbugs-maven-plugin</artifactId>
                <version>${spotbugs.maven.plugin.version}</version>
                <configuration>
                    <effort>Max</effort>
                    <threshold>Default</threshold>
                    <xmlOutput>true</xmlOutput>
                    <findbugsXmlOutput>true</findbugsXmlOutput>
                    <findbugsXmlWithMessages>true</findbugsXmlWithMessages>
                    <failOnError>true</failOnError>
                </configuration>
                <executions>
                    <execution>
                        <phase>verify</phase>
                        <goals>
                            <goal>check</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>
        </plugins>
    </build>
</project>
```

`src/main/java/com/example/NonNullHttpMessageConverter.java`:

```java
package com.example;

import java.io.IOException;
import java.util.List;

import org.slf4j.LoggerFactory;
import org.springframework.http.HttpInputMessage;
import org.springframework.http.HttpOutputMessage;
import org.springframework.http.MediaType;
import org.springframework.http.converter.HttpMessageConverter;
import org.springframework.http.converter.HttpMessageNotReadableException;
import org.springframework.http.converter.HttpMessageNotWritableException;

/**
 * A {@link MarshallingHttpMessageConverter} that throws an exception and logs the HTTP response, instead of just returning
 * null, if the payload is empty.
 */
public class NonNullHttpMessageConverter<T> implements HttpMessageConverter<T> {

    private final HttpMessageConverter<T> converter;

    public NonNullHttpMessageConverter(HttpMessageConverter<T> converter) {
        this.converter = converter;
    }

    @Override
    public boolean canRead(Class<?> clazz, MediaType mediaType) {
        return this.converter.canRead(clazz, mediaType);
    }

    @Override
    public boolean canWrite(Class<?> clazz, MediaType mediaType) {
        return this.converter.canWrite(clazz, mediaType);
    }

    @Override
    public List<MediaType> getSupportedMediaTypes() {
        return this.converter.getSupportedMediaTypes();
    }

    @Override
    public T read(Class<? extends T> clazz, HttpInputMessage message) throws IOException, HttpMessageNotReadableException {
        final T obj = this.converter.read(clazz, message);
        if (obj == null) {
            LoggerFactory.getLogger(this.getClass()).error("rec'd unexpected empty HTTP response:\n{}", message);
            throw new HttpMessageNotReadableException("unexpected empty response", message);
        }
        return obj;
    }

    @Override
    public void write(T t, MediaType contentType, HttpOutputMessage message) throws IOException, HttpMessageNotWritableException {
        this.converter.write(t, contentType, message);
    }
}
```

Maven output:
```
$ mvn clean verify
[INFO] Scanning for projects...
[INFO] 
[INFO] --------------------------< com.example:bug >---------------------------
[INFO] Building Bug Example 1.0
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ bug ---
[INFO] Deleting /Users/archie/bugs/spotbugs-0001/target
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ bug ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /Users/archie/bugs/spotbugs-0001/src/main/resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.7.0:compile (default-compile) @ bug ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 1 source file to /Users/archie/bugs/spotbugs-0001/target/classes
[INFO] 
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ bug ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /Users/archie/bugs/spotbugs-0001/src/test/resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.7.0:testCompile (default-testCompile) @ bug ---
[INFO] No sources to compile
[INFO] 
[INFO] --- maven-surefire-plugin:2.12.4:test (default-test) @ bug ---
[INFO] No tests to run.
[INFO] 
[INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ bug ---
[INFO] Building jar: /Users/archie/bugs/spotbugs-0001/target/bug-1.0.jar
[INFO] 
[INFO] >>> spotbugs-maven-plugin:4.5.0.0:check (default) > :spotbugs @ bug >>>
[INFO] 
[INFO] --- spotbugs-maven-plugin:4.5.0.0:spotbugs (spotbugs) @ bug ---
[INFO] Fork Value is true
     [java] [main] WARN edu.umd.cs.findbugs.TextUICommandLine - -output option and -outputFile option are deprecated. Set file path to each option for reporter.
[INFO] Done SpotBugs Analysis....
[INFO] 
[INFO] <<< spotbugs-maven-plugin:4.5.0.0:check (default) < :spotbugs @ bug <<<
[INFO] 
[INFO] 
[INFO] --- spotbugs-maven-plugin:4.5.0.0:check (default) @ bug ---
[INFO] BugInstance size is 2
[INFO] Error size is 0
[INFO] Total bugs: 2
[ERROR] Medium: new com.example.NonNullHttpMessageConverter(HttpMessageConverter) may expose internal representation by storing an externally mutable object into NonNullHttpMessageConverter.converter [com.example.NonNullHttpMessageConverter] At NonNullHttpMessageConverter.java:[line 24] EI_EXPOSE_REP2
[ERROR] Medium: Redundant nullcheck of obj, which is known to be non-null in com.example.NonNullHttpMessageConverter.read(Class, HttpInputMessage) [com.example.NonNullHttpMessageConverter] Redundant null check at NonNullHttpMessageConverter.java:[line 45] RCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE
[INFO] 


To see bug detail using the Spotbugs GUI, use the following command "mvn spotbugs:gui"



[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  2.728 s
[INFO] Finished at: 2022-01-03T14:59:36-06:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.github.spotbugs:spotbugs-maven-plugin:4.5.0.0:check (default) on project bug: failed with 2 bugs and 0 errors  -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

The `RCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE` warning is bogus.

Note: if you remove the `LoggerFactory.getLogger...` line then the warning goes away.
